03e_more_ipyautoui_pydantic.ipynb

```diff
@@ -152,46 +168,209 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "ui_data_smooth = AutoUi(DataSelector)\n",
-    "ui_data_smooth"
+    "controls = AutoUi(DataSelector)\n",
+    "controls"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8d168a22-8652-4fbc-b26d-45cb213d43d1",
+   "metadata": {},
+   "source": [
+    "### Step 3: Enforce the constraint between `window_size` and `polynomial_order`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b4219527-b183-4cd2-839d-8660e76ad646",
+   "metadata": {},
+   "source": [
+    "In an earlier notebook we constrained the smoothing settings so that the user could not set an invalid value. We do that again below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "02408499-6ccb-46dc-9959-fce463330962",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dashboard_pydantic.widgets_autoui import make_enforcer\n",
```

Comment:
Probably should use the key version of the module, since you did above:
```suggestion
    "from key.dashboard_pydantic.widgets_autoui import make_enforcer\n",
```